### ISSUE TITLE
Stream with wildcard and apparently incompatible filter type

### ISSUE THREAD

**Issue by blablubbabc** [typetools/checker-framework#5064]
Hello,
I am currently try to resolve some of the problems CheckerFrameworks complains about after I annotated my code base for null checking.
I am using the Gradle plugin (v 0.6.8) to invoke the CheckerFramework (v 3.21.2).

I have encountered this pattern in my code base which compiles fine, but is flagged by CheckerFramework:
```
List<? extends @NonNull String> list = new ArrayList<>(Arrays.asList("bla"));
Stream<? extends @NonNull String> stream = list.stream(); // This is what I got as starting point, String is an example type
Predicate<@Nullable Object> filter = (obj) -> true;
Predicate<? super @NonNull String> filter2 = filter; // This is the type of predicate I try to use
long count = stream
				.filter(filter2) // Error.
				.count();
```

Using `filter` instead of `filter2` works. But why is `filter2` not accepted here by CF as argument for `Stream#filter`?

CF error:
```
[argument] incompatible argument for parameter arg0 of filter.

				.filter(filter2) // Error
				        ^
  found   : @UnknownKeyFor Predicate<capture#282[ extends @UnknownKeyFor Object super @UnknownKeyFor String]>

  required: @UnknownKeyFor Predicate<?[ extends @UnknownKeyFor Object super capture#950[ extends @UnknownKeyFor String super @KeyForBottom Void]]>
```

Without the wildcard in the Stream, the error is gone, but this requires an unchecked cast:
```
@SuppressWarnings("unchecked")
Stream<@NonNull String> stream = (Stream<@NonNull String>) list.stream();
```
